tests/test_iri.py

```diff
@@ -0,0 +1,148 @@
+
+from aatoolbox.datasources.iri.iri_seasonal_forecast import (
+    _MODULE_BASENAME,
+    IriForecastDominant,
+    IriForecastProb,
+)
+from aatoolbox.utils.geoboundingbox import GeoBoundingBox
+
+FAKE_AA_DATA_DIR = "fake_aa_dir"
+ISO3 = "abc"
+
+
+@pytest.fixture(scope="session", autouse=True)
+def mock_aa_data_dir(session_mocker):
+    """Mock out the AA_DATA_DIR environment variable."""
+    session_mocker.patch.dict(
+        "aatoolbox.config.pathconfig.os.environ",
+        {"AA_DATA_DIR": FAKE_AA_DATA_DIR},
+    )
+
+
+@pytest.fixture()
+def mock_download(mocker):
+    """
+    Call download with mocked _download.
+
+    `forecast_type` is the type of forecast to
+    test, can be either 'prob' or 'dominant'.
+    """
+    download_mock = mocker.patch(
+        "aatoolbox.datasources.iri."
+        "iri_seasonal_forecast._IriForecast._download"
+    )
+
+    def _mock_download(forecast_type):
+        geo_bounding_box = GeoBoundingBox(north=6, south=3.2, east=-2, west=3)
+        if forecast_type == "prob":
+            iri_class = IriForecastProb(
+                iso3=ISO3, geo_bounding_box=geo_bounding_box
+            )
+        elif forecast_type == "dominant":
+            iri_class = IriForecastDominant(
+                iso3=ISO3, geo_bounding_box=geo_bounding_box
+            )
+        iri_class.download(iri_auth=None)
+        _, kwargs_download = download_mock.call_args
+        url = kwargs_download["url"]
+        filepath = kwargs_download["filepath"]
+        return url, filepath
+
+    return _mock_download
+
+
+def test_download_call_prob(mock_download):
+    """Test download for tercile probability forecast."""
+    url, filepath = mock_download("prob")
+    assert url == (
+        "https://iridl.ldeo.columbia.edu/SOURCES/.IRI/.FD/"
+        ".NMME_Seasonal_Forecast/"
+        ".Precipitation_ELR/.prob/X/%283.0%29%28-2.0%29RANGEEDGES/"
+        "Y/%286.0%29%283.0%29RANGEEDGES/data.nc"
+    )
+
+    assert filepath == (
+        Path(FAKE_AA_DATA_DIR) / f"private/raw/{ISO3}/{_MODULE_BASENAME}/"
+        f"abc_iri_forecast_seasonal_precipitation_"
+        f"tercile_prob_Np6Sp3Em2Wp3.nc"
+    )
+
+
+def test_download_call_dominant(mock_download):
+    """Test download for dominant tercile forecast."""
+    url, filepath = mock_download("dominant")
+    assert url == (
+        "https://iridl.ldeo.columbia.edu/SOURCES/.IRI/.FD/"
+        ".NMME_Seasonal_Forecast/"
+        ".Precipitation_ELR/.dominant/X/%283.0%29%28-2.0%29RANGEEDGES/"
+        "Y/%286.0%29%283.0%29RANGEEDGES/data.nc"
+    )
+
+    assert filepath == (
+        Path(FAKE_AA_DATA_DIR) / f"private/raw/{ISO3}/{_MODULE_BASENAME}/"
+        f"abc_iri_forecast_seasonal_"
+        f"precipitation_tercile_dominant_Np6Sp3Em2Wp3.nc"
+    )
+
+
+def test_process(mocker):
+    """Test process for IRI forecast."""
+    ds = xr.DataArray(
+        np.reshape(a=np.arange(16), newshape=(2, 2, 2, 2)),
+        coords=[
+            [1, 2],
```

Comment:
I know this is silly but if you make each coord a tuple instead of a list, the type checker won't complain ;)

Reply:
Not sure if I fully got what you mean, if I make the lists inside the list tuples, that doesn't work. And which type checker complaining? 

I now made it a dict, which is at least cleaner. Does that do what you meant?